.github/workflows/deps-update.yml

```diff
@@ -0,0 +1,116 @@
+name: Dependency Updates
+
+on:
+  schedule:
+    - cron: '42 5 * * 1' # Mondays at 5:42 AM
+  workflow_dispatch: {} # Manual runs
+
+permissions: read-all
+
+jobs:
+  update-go:
+    name: Update Go Dependencies
+    runs-on: ubuntu-latest
+    outputs:
+      changes-needed: ${{ steps.is-tree-dirty.outputs.result }}
+    env:
+      GOTOOLCHAIN: local # Prohibits adding `toolchain` directives to go.mod files.
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5
+        with:
+          go-version: '1.21' # Minimum supported go release
+          cache-dependency-path: '**/*.sum'
+
+      # Select the latest available version of gopkg.in/DataDog/dd-trace-go.v1, while ignoring all
+      # the `v1.999.*` versions, which are experimental pre-releases.
+      - name: Determine gopkg.in/DataDog/dd-trace-go.v1 version
+        id: dd-trace-go
+        run: |-
+          set -euo pipefail
+          version=$(go list -f '{{range .Versions}}{{.}}{{"\n"}}{{end}}' -m -versions gopkg.in/DataDog/dd-trace-go.v1 | grep -v -E '^v1\.999\.' | tail -n 1)
+          echo "version=${version}" >> "${GITHUB_OUTPUTS}"
+
+      # Passing "go@<version>" to "go get -u" ensures no dependencies get upgraded to a release that
+      # does not support that specific go release.
+      - name: "Update: Project root"
+        run: go get -t -u go@1.21 gopkg.in/DataDog/dd-trace-go.v1@${{ steps.dd-trace-go.outputs.version }} ./... && go mod tidy
+      - name: "Update: Samples"
+        run: go get -t -u go@1.21 gopkg.in/DataDog/dd-trace-go.v1@${{ steps.dd-trace-go.outputs.version }} ./... && go mod tidy
+        working-directory: samples
+      - name: "Update: Integration Tests"
+        run: go get -t -u go@1.21 gopkg.in/DataDog/dd-trace-go.v1@${{ steps.dd-trace-go.outputs.version }} ./... && go mod tidy
+        working-directory: _integration-tests
```

Comment:
Why put the `./...` thingy ?

Reply:
If you don't put `./...` and pass _any_ argument (`package@query`), then `go get -u` will only touch those things you specified and their transitive dependencies... We wanna update everything that's in use.